src/Raven.Client/Documents/Session/ClusterTransactionSession.cs

```diff
@@ -452,6 +470,35 @@ internal bool HasChanged(CompareExchangeValue<BlittableJsonReaderObject> origina
                 return originalValue.Value.Equals(newValue.Value) == false;
             }
 
+            internal bool MetadataHasChanged(IMetadataDictionary oldMetadata, IMetadataDictionary newMetadata)
+            {
+                if (newMetadata == null && oldMetadata == null)
+                {
+                    return false;
+                }
+
+                if (newMetadata != null)
+                {
+                    if (oldMetadata == null)
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        foreach (var key in newMetadata.Keys)
+                        {
+                            newMetadata.TryGetValue(key, out string newVal);
+                            oldMetadata.TryGetValue(key, out string oldVal);
+                            if (oldVal != newVal)
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+                return false;
```

Comment:
What happened if `oldMetadata != null` but `newMetadata == null`?

Reply:
Fixed it